README.md

```diff
@@ -2,3 +2,18 @@
 ![rust workflow](https://github.com/helium/proto/actions/workflows/rust.yml/badge.svg)
 ![node.js workflow](https://github.com/helium/proto/actions/workflows/node.yml/badge.svg)
 ![erlang workflow](https://github.com/helium/proto/actions/workflows/erlang.yml/badge.svg)
+
+## Contributing
+
+- Avoid `float` in Protobufs because representations differ across hardware architectures
+  + There are many floating point representations from IEEE, plus others
+  + i.e., some range of interior digits are random per float spec
+- Frequency should use `uint32`
+  + Locally, define accessor functions to return float after appropriate division
+- rssi or signal is always negative, thus use `sint32`
+- snr is signal-to-noise ratio and should be `uint32`
+- Fetch and share time in nanos, then truncate to appropriate granularity as needed
```

Comment:
`truncate to appropriate granularity`: does this mean 

I've had success using the ["well known type"](https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/timestamp.proto#L133) in other projects and letting clients figure out what to do with the extra precision.

Reply:
we don't share time in nanos in all protos.. some are millis, nanos or secs depending on whether sharing something finer grained makes sense (or is available). The doc string for each timestamp indicates what it is and the rust `file_store::MsgTimestamp` trait can be used to ensure you decode/encode the timestamp from the right units to `DateTime<Utc>`.

We're sort of stuck with this pattern for now since we already have active use of it so introducing partial switchovers doesn't seem right currently

Reply:
staying with this pattern makes sense given this additional context.

Reply:
Not sure if it applicable here, but I wanted to add that some LoRa Concentrators (w/ SX1303) can record fine-timestamp of received packets. The fine-timestamp has nanosecond precision and could be someday useful for calculating Time-Difference-of-Arrival. To perform the TDoA calculation, we need to preserve the nanosecond precision.

Reply:
That's correct and why witness and beacon report timestamps are in nanos